active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,320 @@
+
+### Formatting input 
+
+[Number formatting](https://docs.microsoft.com/en-us/uwp/api/windows.globalization.numberformatting) can be used to format the value of a Numberbox by configuring an instance of a formatting class and assigning it to the `NumberFormatter` property. Decimal, currency, percent, and significant figures are few of the number formatting classes available. 
+
+Here is an example of using DecimalFormatter to format a NumberBox's value to have one integer digit and two fraction digits:  
+
+XAML
+```XAML
+<NumberBox  x:Name="FormattedNumberBox"
+Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}" />
+```
+
+C#
+```C#
+private void SetNumberBoxNumberFormatter()
+{
+    DecimalFormatter formatter = new DecimalFormatter();
+    formatter.IntegerDigits = 1;
+    formatter.FractionDigits = 2;
+    FormattedNumberBox.NumberFormatter = formatter;
+}
+```
+
+![A NumberBox showign a value of 0.00.](images/numberbox-formatted.PNG)
+
+### Enabling hyper scroll 
+
+Setting `IsHyperScrollEnabled` to true enables the value of NumberBox to be incremented or decremented by scrolling when a NumberBox has focus and is being hovered over. 
+
+XAML
+```XAML
+<NumberBox Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}"
+    IsHyperScrollEnabled="True" />
+```
+
+## Remarks
+
+### Input Scope
+
+`Number` will be used for the [input scope](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Input.InputScopeNameValue). This input scope is intended for working with digits 0-9. This may be overwritten but alternative InputScope types will not be explicitly supported. 
+
+### Not a Number
+
+When a NumberBox is cleared of input, `Value` will be set to `NaN` to indicate no numerical value is present. 
+
+## Recommendations
+
+* `Text` and `Value` make it easy to capture the value of a NumberBox as a String or as a Double without needing to convert the value bewteen types. When programmatically altering the value of a NumberBox, it is recommended to do so through the `Value` property. `Value` will overwrite `Text` in initial set up. After the initial set up, changes to one will be progrogated to the other, but consistently making programmatic changes through `Value` helps avoid any conceptual misunderstanding that NumberBox will accept non-numeric characters through `Text`.  
+
+## API Notes
+
+### Notable Properties  
+
+| Name | Description |
+|:-:|:--|
+| StepFrequency | Gets or sets the value part of a value range that steps should be created for. |
+
+## API Details
+<!-- todo: Missing some APIs in here, like TemplatSettings class and event args classes -->
+
+```c++ 
+enum NumberBoxSpinButtonPlacementMode
+{
+    Hidden,
+    Compact,
+    Inline,
+};
+
+enum NumberBoxValidationMode
+{
+    InvalidInputOverwritten,
+    Disabled,
+};
+
+runtimeclass NumberBoxValueChangedEventArgs
+{
+    Double OldValue{ get; };
+    Double NewValue{ get; };
+};
+
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.Control
```

Comment:
Is it interesting to provide configurability of the calculation engine, or make the engine available for general-purpose expression parsing?  So I could eventually, say, plug in something that routed the expression to wolframalpha.com and handed back the resulting value?

Reply:
Discussion: Consider as a future feature.

Reply:
Resolved with [this ](https://github.com/microsoft/microsoft-ui-xaml-specs/commit/005fe284343dd91fdf7a4febe4dc737820527963)commit.